src/components/App/index.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import './styles.css';
+
+import Display from '../Display';
+import ButtonPanel from '../ButtonPanel';
+
+import calculate from '../../services/calculate';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+      calculated: false,
+    }
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(buttonName) {
+    const result = calculate(this.state, buttonName);
+    this.setState(result);
+  }
+
+  getDisplayText() {
```

Comment:
Is there a way we can integrate the logic in `getDisplayText` to our `calculate` method such that our only method inside this class is `handleClick` ?

Reply:
Actually, it's just a simple pure function, gets parameters and always returns the same for same inputs, also it doesn't change any given parameter. So, we can move it wherever we want. But our class is simple enough right now, it's good to see it in here maybe out of the class scope would better. I'm gonna make it like that. If you have another idea, I will be pleasured to hear from you. 


**CHANGED** You can check the related [commit](https://github.com/btuerker/react-calculator/pull/1/commits/920d9bd5ef6acd22fe215090aefd58e86c777302)